emily/handler/src/api/handlers/chainstate.rs

```diff
@@ -101,31 +88,28 @@ pub async fn get_chainstate_at_height(
     operation_id = "setChainstate",
     path = "/chainstate",
     tag = "chainstate",
-    request_body = SetChainstateRequestBody,
+    request_body = Chainstate,
     responses(
         // TODO(271): Add success body.
-        (status = 201, description = "Chainstate updated successfully", body = SetChainstateResponse),
-        (status = 400, description = "Invalid request body"),
-        (status = 404, description = "Address not found"),
-        (status = 405, description = "Method not allowed"),
-        (status = 500, description = "Internal server error")
+        (status = 201, description = "Chainstate updated successfully", body = Chainstate),
+        (status = 400, description = "Invalid request body", body = ErrorResponse),
+        (status = 404, description = "Address not found", body = ErrorResponse),
+        (status = 405, description = "Method not allowed", body = ErrorResponse),
+        (status = 500, description = "Internal server error", body = ErrorResponse)
     )
 )]
-pub async fn set_chainstate(
-    context: EmilyContext,
-    body: SetChainstateRequestBody,
-) -> impl warp::reply::Reply {
+pub async fn set_chainstate(context: EmilyContext, body: Chainstate) -> impl warp::reply::Reply {
     // Internal handler so `?` can be used correctly while still returning a reply.
     async fn handler(
         context: EmilyContext,
-        body: SetChainstateRequestBody,
+        body: Chainstate,
     ) -> Result<impl warp::reply::Reply, Error> {
         // Convert body to the correct type.
         let chainstate: Chainstate = body;
         add_chainstate_entry_or_reorg(&context, &chainstate).await?;
         // Respond.
         Ok(with_status(
-            json(&(chainstate as SetChainstateResponse)),
+            json(&(chainstate as Chainstate)),
```

Comment:
The move on line 108 seems unnecessary, couldn't you just rename the `body` input variable to `chainstate` or just use body throughout?